packages/astro/src/core/config/schema.ts

```diff
@@ -346,11 +348,25 @@ export const AstroConfigSchema = z.object({
 						defaultLocale: z.string(),
 						locales: z.string().array(),
 						fallback: z.record(z.string(), z.string()).optional(),
-						// TODO: properly add default when the feature goes of experimental
-						routingStrategy: z
-							.enum(['prefix-always', 'prefix-other-locales'])
-							.optional()
-							.default('prefix-other-locales'),
+						routing: z
+							.object({
+								prefixDefaultLocale: z.boolean().default(false),
+								strategy: z.enum(['pathname']).default('pathname'),
+							})
+							.default({})
+							.transform((routing) => {
+								let strategy;
+								switch (routing.strategy) {
+									case 'pathname': {
+										if (routing.prefixDefaultLocale === true) {
+											strategy = 'prefix-always';
+										} else {
+											strategy = 'prefix-other-locales';
+										}
+									}
+								}
+								return strategy as RoutingStrategies;
```

Comment:
Nit: can’t we add the type when you declare `strategy` rather than asserting it in the return statement?

```suggestion
								let strategy: RoutingStrategies;
								switch (routing.strategy) {
									case 'pathname': {
										if (routing.prefixDefaultLocale === true) {
											strategy = 'prefix-always';
										} else {
											strategy = 'prefix-other-locales';
										}
									}
								}
								return strategy;
```